src/ontology/ro-edit.owl

```diff
@@ -6140,6 +6143,41 @@ AnnotationAssertion(oboInOwl:inSubset obo:RO_0012008 obo:valid_for_gocam)
 AnnotationAssertion(rdfs:label obo:RO_0012008 "results in fusion of"@en)
 SubObjectPropertyOf(obo:RO_0012008 obo:RO_0002592)
 
+# Object Property: obo:RO_0012009 (constitutively upstream of)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0012009 "p is constitutively upstream of q iff p is causally upstream of q, p is required for execution of q or a part of q, and the execution of p is approximately constant.")
+AnnotationAssertion(oboInOwl:created_by obo:RO_0012009 "pg"^^xsd:string)
+AnnotationAssertion(oboInOwl:creation_date obo:RO_0012009 "2022-09-26T06:01:01Z"^^xsd:dateTime)
+AnnotationAssertion(oboInOwl:inSubset obo:RO_0012009 obo:valid_for_go_annotation_extension)
+AnnotationAssertion(oboInOwl:inSubset obo:RO_0012009 obo:valid_for_gocam)
+AnnotationAssertion(rdfs:label obo:RO_0012009 "constitutively upstream of"@en)
+SubObjectPropertyOf(obo:RO_0012009 obo:RO_0002304)
+
+# Object Property: obo:RO_0012010 (removes input for)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0012010 "p removes input for q iff p is causally upstream of q, there exists some c such that p has_input c and q has_input c, p reduces the levels of c, and c is rate limiting for execution of q.")
+AnnotationAssertion(oboInOwl:created_by obo:RO_0012010 "pg"^^xsd:string)
+AnnotationAssertion(oboInOwl:creation_date obo:RO_0012010 "2022-09-26T06:06:20Z"^^xsd:dateTime)
+AnnotationAssertion(oboInOwl:inSubset obo:RO_0012010 obo:valid_for_go_annotation_extension)
+AnnotationAssertion(oboInOwl:inSubset obo:RO_0012010 obo:valid_for_gocam)
+AnnotationAssertion(rdfs:label obo:RO_0012010 "removes input for"@en)
+SubObjectPropertyOf(obo:RO_0012010 obo:RO_0002305)
+
+# Object Property: obo:RO_0012011 (indirectly causally upstream of)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0012011 "p is indirectly causally upstream of q iff p is causally upstream of q and there exists some process r such that p is causally upstream of r and r is causally upstream of q.")
+AnnotationAssertion(oboInOwl:created_by obo:RO_0012011 "pg"^^xsd:string)
+AnnotationAssertion(oboInOwl:creation_date obo:RO_0012011 "2022-09-26T06:07:17Z"^^xsd:dateTime)
+AnnotationAssertion(rdfs:label obo:RO_0012011 "indirectly causally upstream of"@en)
+SubObjectPropertyOf(obo:RO_0012011 obo:RO_0002411)
+
+# Object Property: obo:RO_0012012 (indirectly regulates)
```

Comment:
needs property chain

Reply:
Based on the text definition, I think this must be a SWRL rule?

```
'indirectly causally upstream of'(?p, ?q), regulates(?p, ?q) -> 'indirectly regulates'(?p, ?q)
```